src/librustdoc/json/conversions.rs

```diff
@@ -172,10 +174,12 @@ crate fn from_def_id(did: DefId) -> Id {
     Id(format!("{}:{}", did.krate.as_u32(), u32::from(did.index)))
 }
 
-fn from_clean_item_kind(item: clean::ItemKind, tcx: TyCtxt<'_>, name: &Option<Symbol>) -> ItemEnum {
+fn from_clean_item(item: clean::Item, tcx: TyCtxt<'_>) -> ItemEnum {
     use clean::ItemKind::*;
-    match item {
-        ModuleItem(m) => ItemEnum::Module(m.into_tcx(tcx)),
+    let name = item.name;
+    let is_crate = item.is_crate();
+    match *item.kind {
+        ModuleItem(m) => ItemEnum::Module(Module { is_crate, items: ids(m.items) }),
```

Comment:
nit: Could you move this `is_crate()` check into the match arm for `ModuleItem`? IIUC that's the only place that it's needed, so I think it's a bit confusing to have it outside that arm.

Reply:
No, that gives a borrow error.
```
error[E0382]: borrow of partially moved value: `item`
   --> src/librustdoc/json/conversions.rs:182:62
    |
182 |         ModuleItem(m) => ItemEnum::Module(Module { is_crate: item.is_crate(), items: ids(m.items) }),
    |                    - value partially moved here              ^^^^ value borrowed here after partial move
    |
    = note: partial move occurs because value has type `types::Module`, which does not implement the `Copy` trait
help: borrow this field in the pattern to avoid moving `item.kind.0`
    |
182 |         ModuleItem(ref m) => ItemEnum::Module(Module { is_crate: item.is_crate(), items: ids(m.items) }),
    |                    ^^^
```

Reply:
Ah, that makes sense.